lib/pbench/test/unit/server/test_cache_manager.py

```diff
@@ -1566,13 +1632,166 @@ def test_lifecycle(
+        assert lock.lock == FakeStream(lockfile)
+        assert not lock.locked
+        assert not lock.exclusive
+        assert lock.unlock
+        assert lock.wait
+        assert len(locks) == 0
+        assert files == [("open", lockfile, "w+")]
+        lock.acquire()
+        assert lock.locked
+        assert locks == [(lockfile, fcntl.LOCK_SH)]
+        lock.release()
+        assert not lock.locked
+        assert files == [("open", lockfile, "w+"), ("close", lockfile)]
+        assert locks == [(lockfile, fcntl.LOCK_SH), (lockfile, fcntl.LOCK_UN)]
+
+        # No-wait
+        reset()
+        lock = LockRef(lockfile, wait=False)
+        assert files == [("open", lockfile, "w+")]
+        assert not lock.wait
+        lock.acquire()
+        assert locks == [(lockfile, fcntl.LOCK_SH | fcntl.LOCK_NB)]
+        lock.release()
+        assert locks == [
+            (lockfile, fcntl.LOCK_SH | fcntl.LOCK_NB),
+            (lockfile, fcntl.LOCK_UN),
+        ]
+        assert files == [("open", lockfile, "w+"), ("close", lockfile)]
+
+        # Exclusive
+        reset()
+        lock = LockRef(lockfile, exclusive=True)
+        assert files == [("open", lockfile, "w+")]
+        assert lock.exclusive
+        lock.acquire()
+        assert locks == [(lockfile, fcntl.LOCK_EX)]
+        lock.release()
+        assert locks == [(lockfile, fcntl.LOCK_EX), (lockfile, fcntl.LOCK_UN)]
+        assert files == [("open", lockfile, "w+"), ("close", lockfile)]
+
+        # Context Manager
+        reset()
+        with LockRef(lockfile) as lock:
+            assert lock.locked
+            assert files == [("open", lockfile, "w+")]
+            assert locks == [(lockfile, fcntl.LOCK_SH)]
+            lock.upgrade()
+            assert locks == [(lockfile, fcntl.LOCK_SH), (lockfile, fcntl.LOCK_EX)]
+            # upgrade is idempotent
+            lock.upgrade()
+            assert locks == [(lockfile, fcntl.LOCK_SH), (lockfile, fcntl.LOCK_EX)]
+            lock.downgrade()
+            assert locks == [
+                (lockfile, fcntl.LOCK_SH),
+                (lockfile, fcntl.LOCK_EX),
+                (lockfile, fcntl.LOCK_SH),
+            ]
+            # downgrade is idempotent
+            lock.downgrade()
+            assert locks == [
+                (lockfile, fcntl.LOCK_SH),
+                (lockfile, fcntl.LOCK_EX),
+                (lockfile, fcntl.LOCK_SH),
+            ]
+        assert locks == [
+            (lockfile, fcntl.LOCK_SH),
+            (lockfile, fcntl.LOCK_EX),
+            (lockfile, fcntl.LOCK_SH),
+            (lockfile, fcntl.LOCK_UN),
+        ]
+        assert files == [("open", lockfile, "w+"), ("close", lockfile)]
+
+        # Context Manager with 'keep' option\
+        reset()
+        with LockRef(lockfile) as lock:
+            assert lock.keep() is lock
+            assert not lock.unlock
+        assert locks == [(lockfile, fcntl.LOCK_SH)]
+        assert files == [("open", lockfile, "w+")]
```

Comment:
Do you want to assert any more of the `lock` state after leaving the `with` block (e.g., `lock.locked`)?